src/core/border_router/routing_manager.cpp

```diff
@@ -1632,6 +1651,86 @@ void RoutingManager::DiscoveredPrefixTable::SignalTableChanged(void)
     mSignalTask.Post();
 }
 
+void RoutingManager::DiscoveredPrefixTable::ProcessNeighborAdvertMessage(
+    const Ip6::Nd::NeighborAdvertMessage &aNaMessage,
+    const Ip6::Address &                  aSrcAddress)
+{
+    Router *router = mRouters.FindMatching(aSrcAddress);
+
+    VerifyOrExit(router != nullptr);
+    VerifyOrExit(aNaMessage.IsValid());
```

Comment:
Should we also validate that the Target Address is the same as what was solicited? This would help ensure that the NA is in response to the NS.

Reply:
Yes. We can check this. Added it in the new push.